src/schema/objects/entities.yaml

```diff
@@ -349,10 +349,13 @@ task:
   name: task
   display_name: Task
   description: |
-    Each task has a unique label that MUST only consist of letters and/or
-    numbers (other characters, including spaces and underscores, are not
-    allowed).
-    Those labels MUST be consistent across subjects and sessions.
+    A set of structured activities performed by the participant.
+    Tasks are usually accompanied by stimuli and responses, and can greatly vary in complexity.
```

Comment:
Is there a distinction to be made between the task definition under common principles and the task entity? This part seems more related to the common principle than the entity.

Reply:
I don't think we should make a distinction unless we want to make things extra confusing.

Are we making such distinctions for subject or session?

Reply:
There are small differences at the moment. The session common principle has a sentence about PET data that the entity definition doesn't have, but I think that's more about information drift between duplicated text.

There _is_ a meaningful difference between the subject common principle and entity, in that the common principle explicitly mentions the term "Participant".

I see what you mean though- perhaps there's no reason to give them distinct definitions.

Reply:
I do think it's worth mentioning the following though:

1. Having the `task` entity present means that having an associated events file is recommended, and there are certain metadata fields that are also recommended or required.
2. It is recommended to use `task-rest` for resting-state scans, and events files are not necessary for resting-state data.

Reply:
>     1. Having the `task` entity present means that having an associated events file is recommended, and there are certain metadata fields that are also recommended or required.
> 
>     2. It is recommended to use `task-rest` for resting-state scans, and events files are not necessary for resting-state data.

I agree that those should be in, but is the "definition" section the best place (not rhetorical). I am ok putting them there but it feels likes this is also blurring the  object and rule delineation we have in the schema, no?

Reply:
Good point. If we find ways to best disentangle objects and rules later we'll cross that bridge when we get to it but for now let's add it as is.